sw/device/tests/otbn_mem_scramble_test.c

```diff
@@ -22,19 +22,59 @@ typedef dif_result_t (*otbn_write_t)(const dif_otbn_t *otbn,
                                      uint32_t offset_bytes, const void *src,
                                      size_t len_bytes);
 
-/**
- * Number of distinct addresses to check in each IMEM and DMEM.
- */
-static const int kNumAddrs = 50;
-
-/**
- * Minimum number of expected integrity errors in IMEM and DMEM after
- * re-scrambling.
- *
- * Note that there is a non-zero chance for every word that the integrity bits
- * after re-scrambling are still valid.
- */
-static const int kNumIntgErrorsThreshold = kNumAddrs - 1;
+enum {
+  /**
+   * Number of distinct addresses to check in each IMEM and DMEM.
+   */
+  kNumAddrs = 50,
+
+  /**
+   * Minimum number of expected integrity errors in IMEM and DMEM after
+   * re-scrambling.
+   *
+   * Note that each word has 7 bits of ECC, so the probability that a word would
```

Comment:
To be more precise, I would rephrase this comment into something like:
`Note that there are 2^32 valid code words and that each non-valid code word triggers an error. Therefore, the probability that a random 39-bit word triggers an error is: (2^39 - 2^32)/ 2^39 = 127/128.`

The rest looks OK to me.